ocaml/xapi/xapi_udhcpd.ml

```diff
@@ -162,7 +162,18 @@ let restart_nolock () =
   let pid = try Unixext.pidfile_read !Xapi_globs.udhcpd_pidfile with _ -> None in
   Opt.iter Unixext.kill_and_wait pid;
   let (_: string * string) = execute_command_get_output !Xapi_globs.busybox [ "udhcpd"; !Xapi_globs.udhcpd_conf ] in
-  ()
+  let start = Mtime_clock.counter () in
+  let rec wait_for_pid n =
+    let now = Mtime_clock.count start in
```

Comment:
Is this correct? `now` is taken at every iteration of the loop. How is this guarantee to fail at some point when there is no upper limit for `n`?